presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -353,7 +348,6 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
                 else {
                     switch (insertExistingPartitionsBehavior) {
                         case APPEND:
-                            checkState(!immutablePartitions);
```

Comment:
undo, this is just defensive programming.

Reply:
I believe this is not good idea to pass immutablePartitions variable through 2 classes just to check it here.
Will not undo.